src/DssGateSuck.sol

```diff
@@ -0,0 +1,166 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity 0.8.11;
+
+interface VatLike {
+    function live() external returns (uint256);
+    function suck(address, address, uint256) external;
+    function dai(address) external view returns (uint256);
+    function move(address, address, uint256) external;
+}
+
+interface JoinLike {
+    function join(address, uint256) external;
+}
+
+/**
+ @title Gate Suck
+ FEATURES
+  * token approval style draw limit on vat.suck
+
+ DEPLOYMENT
+  * ideally, each gate contract should only be linked to a single integration
+  * authorized integration can then request a dai amount from gate contract with a "suck" call
+
+ DRAW LIMIT
+  * a limit on the amount of dai that can an integration can draw with a vat.suck call
+  * simple gate uses an approved total amount, similar to a token approval
+  * integrations can access up to this dai amount in total
+  * if they repay the dai they can "refill the approval amount" allowing for future draws.
+
+ DAI FORMAT
+  * integrates with dai balance on vat, which uses the dsmath rad number type- 45 decimal fixed-point number
+
+ MISCELLANEOUS
+  * does not execute vow.heal to ensure the dai draw amount from vat.suck is lower than the surplus buffer currently held in vow
+  * vat, and vow addresses cannot be updated after deployment
+*/
+contract DssGateSuck {
+    // --- Auth ---
+    mapping (address => uint256) public wards;                                       // Addresses with admin authority
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    function rely(address _usr) external auth { wards[_usr] = 1; emit Rely(_usr); }  // Add admin
+    function deny(address _usr) external auth { wards[_usr] = 0; emit Deny(_usr); }  // Remove admin
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssGateSuck/not-authorized");
+        _;
+    }
+
+    // --- Integration Access Control ---
+    mapping (address => uint256) public can;
+    event Hope(address usr);
+    event Nope(address usr);
+    function hope(address _a) external auth {
+        require(_a != address(0), "DssGateSuck/cannot-hope-nothing");
+        can[_a] = 1;
+        emit Hope(_a);
+    }
+    function nope(address _a) external auth {
+        can[_a] = 0;
+        emit Nope(_a);
+    }
+    modifier wish { require(can[msg.sender] == 1, "DssGateSuck/bud-not-authorized"); _; }
+
+    /// maker protocol vat
+    address public immutable vat;
+    /// maker protocol vow
+    address public immutable vow;
```

Comment:
IMO `vow` shouldn't be immutable as it might be updated. We keep it always upgradable.